### PR TITLE
Switches over the Contextify Passes to use the Structure generator

### PR DESCRIPTION
**Pull request by mariari**
We should not be matching on the S-expression sturcture directly as we did in this file, this pr fixes that mistake.

Finishing the rest of Desugar will also be in the PR, in the second commit.

### CODE REVIEW

**Review by Acentelles — APPROVED** [review]

**Comment by cyberglot** [comment]
I'd need to update my stuff as well. Thanks, it looks way better now.

File: library/Translate/src/Juvix/Sexp/Structure.hs [hunk 1]
```diff
@@ -702,3 +856,185 @@ toRecordNoPunned form
...
+
+isDefModule :: Sexp.T -> Bool
+isDefModule (Sexp.Cons form _) = Sexp.isAtomNamed form nameDefModule
+isDefModule _ = False
+
+toDefModule :: Sexp.T -> Maybe DefModule
+toDefModule form
+  | isDefModule form =
+    case form of
+      _DefModule Sexp.:> sexp1 Sexp.:> sexp2 Sexp.:> sexp3 ->
```
Comment (Acentelles) [inline library/Translate/src/Juvix/Sexp/Structure.hs:1005]: This _DefModule is a weird name. It refers to the name I think. Maybe prepending name to it? E.g. _nameDefModule?

File: library/Translate/test/Test/Desugar/Sexp.hs [hunk 2]
```diff
@@ -234,6 +256,6 @@ modLetWorkAsExpected =
         "(:defun foo ()\
         \   (let foo ()\
         \         (let bar () 3 \
-        \            (:let-type foo (() (XTZ))\
```
Comment (Acentelles) [inline library/Translate/test/Test/Desugar/Sexp.hs:237]: Was it working before as expected?

File: library/Translate/src/Juvix/Sexp/Structure.hs (hunk 1 above)
Comment (mariari) [reply library/Translate/src/Juvix/Sexp/Structure.hs:1005]: Do you want this to be true for all names or just this one?

File: library/Translate/test/Test/Desugar/Sexp.hs (hunk 2 above)
Comment (mariari) [reply library/Translate/test/Test/Desugar/Sexp.hs:237]: I'd argue no, as the form is harder to deal with in general, as the car is arguments yet the cdr is a bunch of sum types, it's just odd as is.

File: library/Translate/src/Juvix/Sexp/Structure.hs (hunk 1 above)
Comment (Acentelles) [reply library/Translate/src/Juvix/Sexp/Structure.hs:1005]: All names. I guess it'd be a small change as it's autogenerated

File: library/Translate/test/Test/Desugar/Sexp.hs (hunk 2 above)
Comment (mariari) [reply library/Translate/test/Test/Desugar/Sexp.hs:237]: Yes, I understand, the form of these aren't fully fit for purpose yet, the :module and :let-mod forms i think need a bit of tweaking. 

I'll likely have to redesign a whole lot when the syntax changes come in (the tests get the brunt of the annoying work, while the code should stay the same!)

File: library/Translate/src/Juvix/Sexp/Structure.hs (hunk 1 above)
Comment (mariari) [reply library/Translate/src/Juvix/Sexp/Structure.hs]: Those changes are now in

**Pull request merged**